src/app/application.cpp

```diff
@@ -278,7 +279,7 @@ void Application::torrentFinished(BitTorrent::TorrentHandle *const torrent)
         program.replace("%T", torrent->currentTracker());
         program.replace("%I", torrent->hash());
 
-        QProcess::startDetached(program);
+        std::thread ([program]() { return system(program.toStdString().c_str()); }).detach();
```

Comment:
I tested your example code. It confirmed my knowledge about threads: process waits until all its threads finished.